### PR TITLE
FEAT: Docs upgrades throughout :chart_with_upwards_trend: 

### PR DESCRIPTION
**Pull request by daquinteroflex**
Includes:

- [X] Better landing page initial
- [x] Better installation instruction page
- [x] Tabbed instructions for multiple platforms with emojis
- [X] Fixes a function wrapping docs issue Lucas raised

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -13,44 +14,87 @@ Tidy3D Electromagnetic Solver
...
-.. * open simple example in colab with API saved as environment variable and `!pip install tidy3d` in the first line.
-.. * toggle between command line - notebook / python instructions in section 1
 Tidy3D is a software package for solving extremely large electrodynamics problems using the finite-difference time-domain (FDTD) method. It can be controlled through either an `open source python package <https://github.com/flexcompute/tidy3d>`_ or a `web-based graphical user interface <https://tidy3d.simulation.cloud>`_.
 
-If you’d rather skip installation and run an example in one of our web-hosted notebooks, `click here <https://tidy3d.simulation.cloud/notebook?file=StartHere.ipynb>`_.
+Get Started
+===========
 
-.. `TODO: open example in colab <https://github.com/flexcompute/tidy3d>`_
+.. tabs::
```
Comment (tylerflex) [inline docs/index.rst:28]: this is great!

File: docs/index.rst [hunk 2]
```diff
@@ -13,44 +14,87 @@ Tidy3D Electromagnetic Solver
...
+
+    .. group-tab:: On the Cloud |:cloud:|
```
Comment (tylerflex) [inline docs/index.rst:30]: maybe "In the Cloud"

File: docs/index.rst [hunk 3]
```diff
@@ -13,44 +14,87 @@ Tidy3D Electromagnetic Solver
...
+
+        If you’d rather skip installation and run an example in one of our web-hosted notebooks, `click here <https://tidy3d.simulation.cloud/notebook?file=StartHere.ipynb>`_.
+
+    .. group-tab:: Install on Linux |:penguin:| & MacOS |:apple:|
```
Comment (tylerflex) [inline docs/index.rst:34]: I think this should be the default tab, most people probably just want to find this `pip install` command so I think it should be immediately visible.

File: docs/install.rst [hunk 4]
```diff
@@ -14,25 +11,37 @@ Before using Tidy3D, you must first `sign up <https://tidy3d.simulation.cloud/si
...
+            pip install --user tidy3d
+            tidy3d configure --apikey=XXX
+
+    .. group-tab:: Any Hosted Environment |:cloud:|
+
+        Where ``XXX`` is your API key, which can be copied from your `account page <https://tidy3d.simulation.cloud/account>`_ in the web interface.
+
+        In a hosted jupyter notebook environment (eg google colab), it may be more convenient to install and configure via the following lines at the top of the notebook.
+
+        .. code-block:: bash
```
Comment (tylerflex) [inline docs/install.rst:37]: should be python syntax highlighting

File: docs/install.rst [hunk 5]
```diff
@@ -43,132 +52,125 @@ To test that the configuration is working from the python browser, you can perfo
...
+.. tabs::
 
-The API key must be in a file called ``.tidy3d/config`` located in your home directory, with the following contents
+    .. group-tab:: Conda/Mamba |:snake:|
 
-.. code-block:: bash
+        If you already have Python installed on your computer, it is possible that some packages in your current environment could have version conflicts with Tidy3D. To avoid this, we strongly recommend that you create a clean Python virtual environment to install Tidy3D.
 
-    apikey = "XXX"
+        We recommend using the Conda package management system to manage your Python virtual environment as well as installing Tidy3D. You can install Conda conveniently via `Anaconda <https://www.anaconda.com/>`__.
```
Comment (tylerflex) [inline docs/install.rst:138]: hm, I guess we do say this in our current installation instructions. Although I've personally found conda a bit annoying to use, although I dont remember the specific details about it now. What are your thoughts on conda? maybe it's the best for windows users but not sure about for everyone else?

**Review by tylerflex — CHANGES_REQUESTED** [review]
Thanks @daquinteroflex I think this really improves things. Just a few minor comments.

**Comment by tylerflex** [comment]
one thing I fogot to mention: I think it's nice that we have some explanation of how to get to 2.6 if the link is broken. But is there a way to insert this information into a custom 404 page? (page not found.  If you were trying to switch from an older version to 2.6, the link may have changed, please navigate from the main page).

File: docs/install.rst (hunk 4 above)
Comment (daquinteroflex) [reply docs/install.rst:37]: My bad fixed them all

Comment (tylerflex) [reply docs/install.rst:37]: a lot of them were actually supposed to be bash, only noticed this one.

File: docs/install.rst (hunk 5 above)
Comment (daquinteroflex) [reply docs/install.rst:138]: We could update it to mamba which is better.  Conda on it's own is old and not great, and very slow. 

Maybe it's  a common enough tool that we should provide instructions! It's good for notebooks and people tinkering, but for reproducible development, poetry has been much better.

**Comment by daquinteroflex** [comment]
Yep so the 404 is in my list in the https://github.com/flexcompute/tidy3d/issues/1148 but will get to that in another PR

**Comment by daquinteroflex** [comment]
Will be probably ready to merge when these changes build btw (should be as just updating submodule) @tylerflex , the next set of docs upgrades will come in the next PR including the 404 if that sounds good

**Review by tylerflex — APPROVED** [review]
ok sounds good to me, feel free to merge whenever the build looks good.

**Pull request merged**